oot.sh

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/bash
+
+while getopts ':c:' OPT; do
+        case ${OPT} in
+        c)
+		case ${OPTARG} in
+		clang|gcc|linaro_gcc)
+			export _compiler=${OPTARG}
+			;;
+		*)
+			echo "Compiler '${OPTARG}' unknown or not implemented"
+			exit
+			;;
+		esac
+		;;
+        ?)
+                echo && echo "Incorrect option: ${OPT}"
+                exit
+                ;;
+        esac
+done
+
+shift $(( OPTIND - 1 ))
+
+
+if (( "$#" == 0 )); then
+	echo "Usage: $0 <device> (Example for akatsuki: $0 akatsuki)"
+	exit
+fi
+
+
+# Default to clang
+if [[ -z "${_compiler}" ]]; then
+	export _compiler=clang
+fi
+
+
+# Platform common
+export BOARD_KERNEL_BASE=0x00000000
+export BOARD_KERNEL_PAGESIZE=4096
+export BOARD_KERNEL_TAGS_OFFSET=0x01E00000
+export BOARD_RAMDISK_OFFSET=0x02000000
+
+for _device in "$@"; do
+	unset BOARD_KERNEL_CMDLINE _platform _has_dtbo _recovery_ramdisk
+
+	BOARD_KERNEL_CMDLINE+=" lpm_levels.sleep_disabled=1"
+	BOARD_KERNEL_CMDLINE+=" service_locator.enable=1"
+
+	# Device specific
+	case ${_device} in
+	mermaid)
+		export _platform=ganges
+		;;
+	discovery|pioneer|voyager)
+		export _platform=nile
+		;;
+	bahamut|griffin)
+		export _platform=kumano
+		;;
+	pdx201)
+		export _platform=seine
+		;;
+	akari|akatsuki|apollo)
+		export _platform=tama
+		;;
+	lilac|maple|poplar)
+		export _platform=yoshino
+		;;
+	*)
+		echo "Device '${_device}' unknown or not implemented"
+		exit
+		;;
+	esac
+
+
+	# Platform specific
+	case ${_platform} in
+#	nile|ganges)
+#		_verity_file=build/target/product/security/verity.x509.pem
+#		_verity_key_id=$(openssl x509 -in $_verity_file -text | grep keyid | sed 's/://g' | tr -d '[:space:]' | tr '[:upper:]' '[:lower:]' | sed 's/keyid//g')
+#
+#		BOARD_KERNEL_CMDLINE+=" veritykeyid=id:$_verity_key_id"
+#		export BOARD_KERNEL_CMDLINE
+#		;;
+	kumano)
+		export _has_dtbo=true
+
+		BOARD_KERNEL_CMDLINE+=" androidboot.bootdevice=1d84000.ufshc"
+		BOARD_KERNEL_CMDLINE+=" swiotlb=2048"
+		BOARD_KERNEL_CMDLINE+=" msm_drm.blhack_dsi_display0=dsi_panel_somc_${_platform}_cmd:config0"
+		export BOARD_KERNEL_CMDLINE
+		;;
+	seine)
+		export _has_dtbo=true
+		export _recovery_ramdisk=false
+
+		BOARD_KERNEL_CMDLINE+=" androidboot.bootdevice=1d84000.ufshc"
+		BOARD_KERNEL_CMDLINE+=" swiotlb=2048"
```

Comment:
That looks like my copypaste mistake, Seine doesn't have `swiotlb` set in `PlatformConfig.mk`.

Reply:
You're right. Fixed.

I've also moved `BOARD_KERNEL_CMDLINE+=" androidboot.hardware=${_device}"` in the `# Device specific` section to honor what SODP devices trees do.

Reply:
It's above the device specific section now... Where it belongs as it's still a generic flag filled in with a specific variable :)